components/script/dom/webglrenderingcontext.rs

```diff
@@ -511,6 +511,132 @@ impl WebGLRenderingContext {
             _ => false,
         }
     }
+    
+    // Remove all the different types of comments from a given shader source
```

Comment:
do you have spec links for this?

Reply:
@Manishearth This is mostly a direct translation of the Firefox version of this function, but I did go through *a* spec (I am not sure if it's *the* spec - based on my limited knowledge of the WebGL standard, it only accepts ES1.0 GLSL) to validate my logic:
https://www.khronos.org/files/opengles_shading_language.pdf

I'll add a comment pointing to this spec and fix the other issues in my next commit, as I am sure feedback related to the logic is yet to follow. :)

Reply:
I'm confused, this code seems way more complex than the Firefox code, what am I missing?

Reply:
I apologize for the confusion.
The Firefox code, to me at least, is very opaque and obtuse and additionally relies on pointer arithmetic. My code tries to accomplish the exact same objective while being transparent enough to ensure full accessibility to anyone who reads it, at the expense of seeming convoluted, as you rightly point out. If my translation is unacceptable by the standards of this project, then I'd be happy to have this PR discarded. :)

Reply:
Nah your code is fine I was just making sure I wasn't missing something because you said it's a direct translation, you are right that it's better to avoid pointer arithmetics. :)